src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -1209,6 +1214,14 @@ trait ChoosesMode {
         parkingResponses
       ) ++ rideHail2TransitIinerary.toVector
 
+      choosesModeData.personData.currentTourMode match {
+        case Some(mode) if mode.isTransit && !combinedItinerariesForChoice.exists(_.tripClassifier == mode) =>
```

Comment:
To help with readability and understanding why this is considered a failed event I suggest rename `mode` to `expectedMode` - or maybe turn the `exists` into a `def` to control the readability?